torchaudio/transforms.py

```diff
@@ -647,18 +651,36 @@ class Resample(torch.nn.Module):
     """
 
     def __init__(self,
-                 orig_freq: int = 16000,
-                 new_freq: int = 16000,
+                 orig_freq: float = 16000,
+                 new_freq: float = 16000,
                  resampling_method: str = 'sinc_interpolation',
                  lowpass_filter_width: int = 6,
                  rolloff: float = 0.99) -> None:
         super(Resample, self).__init__()
-        self.orig_freq = orig_freq
-        self.new_freq = new_freq
+
+        if not (int(orig_freq) == orig_freq and int(new_freq) == new_freq):
+            warnings.warn(
+                "Non-integer frequencies are being cast to ints and may result in poor resampling quality "
+                "because the underlying algorithm requires an integer ratio between `orig_freq` and `new_freq`. "
+                "Using non-integer valued frequencies will throw an error in the next release. "
+                "To work around this issue, manually convert both frequencies to integer values "
+                "that maintain their resampling rate ratio before passing them into the function "
+                "Example: To downsample a 44100 hz waveform by a factor of 8, use "
+                "`orig_freq=8` and `new_freq=1` instead of `orig_freq=44100` and `new_freq=5512.5` "
+                "For more information or to leave feedback about this change, please refer to "
+                "https://github.com/pytorch/audio/issues/1487."
+            )
```

Comment:
Copy-pasting the chunk is cumbersome for the future maintenance. Instead, this logic of GCD and integer validation should go inside of `_get_sinc_resample_kernel`. That way, one code is used from different entry points.
In future, when we want to add more variations of kernels, we can extract the GCD and integer validation and make one helper function, which is called from `_get_sinc_resample_kernel` and the new kernel function.